internal/integration/client_test.go

```diff
@@ -818,6 +819,87 @@ func TestClient_BulkWrite(t *testing.T) {
 			})
 		}
 	})
+	mt.RunOpts("bulk write with bypassEmptyTsReplacement", mtBulkWriteOpts, func(mt *mtest.T) {
+		mt.Parallel()
+
+		newOpts := func(option bson.D) *options.ClientBulkWriteOptionsBuilder {
+			opts := options.ClientBulkWrite()
+			err := xoptions.SetInternalClientBulkWriteOptions(opts, "addCommandFields", option)
+			require.NoError(mt, err, "unexpected error: %v", err)
+			return opts
+		}
+
+		marshalValue := func(val interface{}) bson.RawValue {
+			t.Helper()
+
+			valType, data, err := bson.MarshalValue(val)
+			require.Nil(t, err, "MarshalValue error: %v", err)
+			return bson.RawValue{
+				Type:  valType,
+				Value: data,
+			}
+		}
+
+		models := []struct {
+			name  string
+			model mongo.ClientWriteModel
+		}{
+			{
+				"insert one",
+				mongo.NewClientInsertOneModel().SetDocument(bson.D{{"x", 1}}),
+			},
+			{
+				"update one",
+				mongo.NewClientUpdateOneModel().SetFilter(bson.D{{"x", 1}}).SetUpdate(bson.D{{"$set", bson.D{{"x", 3.14159}}}}),
+			},
+			{
+				"update many",
+				mongo.NewClientUpdateManyModel().SetFilter(bson.D{{"x", 1}}).SetUpdate(bson.D{{"$set", bson.D{{"x", 3.14159}}}}),
+			},
+			{
+				"replace one",
+				mongo.NewClientReplaceOneModel().SetFilter(bson.D{{"x", 1}}).SetReplacement(bson.D{{"x", 3.14159}}),
+			},
+		}
+
+		testCases := []struct {
+			name     string
+			opts     *options.ClientBulkWriteOptionsBuilder
+			expected bson.RawValue
+		}{
+			{
+				"empty",
+				options.ClientBulkWrite(),
+				bson.RawValue{},
+			},
```

Comment:
Optional: Keyed struct literals are easier to use. Consider including the key names.